pkg/defkinds/php/builder_test.go

```diff
@@ -78,10 +80,38 @@ func initBuildLLBForProdStageTC(t *testing.T, mockCtrl *gomock.Controller) build
 	}
 }
 
+func initBuildLLBForWebserverProdStageTC(t *testing.T, mockCtrl *gomock.Controller) buildTC {
+	genericDef := loadGenericDef(t, "testdata/build/with-webserver.yml", "testdata/build/with-webserver.lock")
+
+	solver := mocks.NewMockStateSolver(mockCtrl)
+	solver.EXPECT().FromBuildContext(gomock.Any()).Times(1)
+	solver.EXPECT().ReadFile(
+		gomock.Any(), "composer.lock", gomock.Any(),
+	).Return([]byte{}, statesolver.FileNotFound)
+
+	kindHandler := php.NewPHPHandler()
```

Comment:
both php kind handler and nginx kind handler are able to build a web server ?

isn't it confusing ?

Reply:
Indeed, it's a bit confusing :confused: But unfortunately until [this](https://github.com/NiR-/zbuild/pull/7#discussion_r359078956) is implemented, I don't know how to handle such case :(

Reply:
maybe the phpkind handler could expose a method to return a webserver handler when the `webserver` config is present ?

:/

Reply:
I could reuse the LLB state and image metadata returned by the PHP builder but because the `webserver` config key is part of the specialized PHP definition and not of the generic definition,  I can't do it at an upper level unfortunately (the specialized definition is parsed by the specialized handler) :confused:

Reply:
> because the webserver config key is part of the specialized PHP definition

is it possible for the php handler to use the webserver handler when parsing the webserver definition ?

>  I can't do it at an upper level unfortunately

maybe the zbuild file structure should not have a `kind` root :thinking: (i.e. a unique handler), but should expose multiple kinds (to use multiple handlers) ?
Something closer to the `services` key of docker-compose files ?

This is against the current design tho :/

Reply:
> is it possible for the php handler to use the webserver handler when parsing the webserver definition ?

I'm not sure to understand your question: the webserver handler is already used by the php handler whenever needed (build, update locks, etc...) :thinking: 

> maybe the zbuild file structure should not have a kind root thinking (i.e. a unique handler), but should expose multiple kinds (to use multiple handlers) ?
Something closer to the services key of docker-compose files ?

The only way to remove the `kind` property would be to have a single "syntax provider" (e.g.`# syntax=akerouanton/zbuild-php:v1.0`) per language but this makes things harder for:

1. Target users IMHO (people would have to know/look for what versions are available for each and every syntax provider) ;
2. Me (or any other developer) as this means probably creating separate GH repos, managing separate release cycles, tools, CI, etc... As the tool is still in its early infancy, I'd prefer to avoid that ;

I'm super happy with the current solution but it's directly inspired by what we usually do with Dockerfiles currently: for instance, create a 1st build stage for php/nodejs and then add another one for nginx, to copy assets from the 1st stage. Moreover, zbuildfiles can't be used with docker-compose for now, so I'd prefer to tackle both issues at the same time :wink: